raspios/overlay-rootfs/etc/systemd/system/dashboard.service

```diff
@@ -0,0 +1,17 @@
+[Unit]
+Description=COGIP Dashboard
+
+Wants=network.target
+After=server.service
```

Comment:
Does the dashboard requires the server to be started ?
I guess yes, so you need `Requires=server.service`.

Reply:
Yes, all services running on the beacon need the server.
They all specify `After=server.service`.
What is the difference with `Requires`?

Reply:
Requires and After are not different, they complement each other.
If you only have After, you ensure that you will start Dashboard after Server but Dashboard does not depend on Server. That is, whether Server succeeds or fails to start, Dashboard will start.

If you only use Requires without After, you make sure to start Server every time you start Dashboard but you don't control the timeline. Often they will start at the same time.

So the only way to ensure a controlled timeline and dependency is to use both, Requires and After.

This is what the systemd doc says:
https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html#Wants=

`Note that requirement dependencies do not influence the order in which services are started or stopped. This has to be configured independently with the After= or Before= options. If unit foo.service pulls in unit bar.service as configured with Wants= and no ordering is configured with After= or Before=, then both units will be started simultaneously and without any delay between them if foo.service is activated.`

Requires has lots of variations: Wants, BindsTo, etc...
A site that explains better than the flat systemd documentation:
https://zarak.fr/sre/systemd-dependencies/

But in your case Requires will be very good because if you stop Server, it will also stop Dashboard.

If it's not clear, ping me on Discord, easier.

Reply:
Ok, I have the same understanding!

So I will let this commit as is, but add another commit to add Requires on server in all service files.